### ISSUE TITLE
Error loading DNA: DNA version mismatched, expected 2.1, got 2.2

### ISSUE THREAD

**Issue by Bymeoryou** [EpicGames/MetaHuman-DNA-Calibration#40]
Hello, I am using JSONStreamWriter to transfer DNA to Json. But I meet this problem when I try to covert the pca_rig.dna, can you help me with this?

**Comment by n0phx** [comment]
Hello, could you share the exact snippet of code that you used which reproduces the issue? Thanks!

**Comment by kalebzaki4** [comment]
Certainly! When using `JSONStreamWriter` to convert a `.dna` file like `pca_rig.dna` to JSON, there are several factors to consider. Below are some common steps and tips to help you troubleshoot the issue you're encountering:

### Steps to Convert DNA to JSON

1. **Initialize the DNA Reader and JSON Writer**:
   Make sure you've correctly initialized the `DNAReader` and `JSONStreamWriter` objects.

   ```python
   from dna import DNAReader, JSONStreamWriter

   # Load your DNA file
   dna_file_path = 'path/to/your/pca_rig.dna'
   dna_reader = DNAReader()
   dna_reader.load(dna_file_path)

   # Create a JSON writer
   json_writer = JSONStreamWriter()
   ```

2. **Transfer DNA Data to JSON**:
   Use the JSON writer to write the DNA data into a JSON format.

   ```python
   json_file_path = 'path/to/your/output.json'
   with open(json_file_path, 'w') as json_file:
       json_writer.write(dna_reader, json_file)
   ```

3. **Check for Errors**:
   If you encounter an error during the conversion process, make sure to check:
   - The path to the `.dna` file is correct.
   - The `.dna` file is not corrupted.
   - You have the necessary permissions to read the file and write to the output location.

### Common Issues and Solutions

- **Module Import Errors**:
  Ensure you have imported the necessary classes and that your environment is correctly set up to include the `dna` module.

- **File Not Found or Access Denied**:
  Verify that the file paths used in your code are correct and that the files exist.

- **Unexpected Format or Corrupted DNA File**:
  If the DNA file is corrupted or not in the expected format, you may need to regenerate the `.dna` file or check its integrity.

- **JSON Writer Exceptions**:
  If the `JSONStreamWriter` throws an exception, check the documentation for the method you're using and ensure you're providing all required parameters.

### Example Code

Here’s a complete example to illustrate the process:

```python
from dna import DNAReader, JSONStreamWriter

# Load your DNA file
dna_file_path = 'path/to/your/pca_rig.dna'
dna_reader = DNAReader()
if not dna_reader.load(dna_file_path):
    print(f"Failed to load DNA file: {dna_file_path}")

# Create a JSON writer
json_writer = JSONStreamWriter()

# Write to JSON
json_file_path = 'path/to/your/output.json'
try:
    with open(json_file_path, 'w') as json_file:
        json_writer.write(dna_reader, json_file)
    print(f"Successfully converted {dna_file_path} to {json_file_path}")
except Exception as e:
    print(f"An error occurred: {e}")
```

### If Problems Persist
If you’re still experiencing issues, please provide the specific error message you receive when trying to convert the `.dna` file. This will help in diagnosing the problem more accurately.

**Comment by chadmv** [comment]
I get this error with the following:

```
from dna_viewer import DNA
dna = DNA(dna_path)
```

This results in 

```
#   File "...\MetaHuman_DNA_Calibration\1.2.0\python-3.9\dna_viewer\dnalib\dnalib.py", line 27, in __init__
#     self.reader = self.create_reader(dna_path)
#   File "...\MetaHuman_DNA_Calibration\1.2.0\python-3.9\dna_viewer\dnalib\dnalib.py", line 52, in create_reader
#     raise RuntimeError(f"Error loading DNA: {status.message}")
# RuntimeError: Error loading DNA: DNA version mismatched, expected 2.1, got 2.3
```
